### ISSUE TITLE
netmiko.ssh_exception.NetmikoTimeoutException: Timed-out reading channel, data not avaible

### ISSUE THREAD

**Issue by xSerioUsx78** [ktbyers/netmiko#2317]
when i want to change the hostname multiple times give me this error.

example:

    device_info = {my info goes here}
    connect = ConnectHandler(**device_info)

    for i in range(5):
          config_set = [f'hostname test{i}']
          connect.send_config_set(config_set)

where is the problem?
is it can't handle this form of commands or should i set some additional attribiute here?

**Comment by ktbyers** [comment]
@xSerioUsx78  Can you post your exception stack trace?

**Comment by xSerioUsx78** [comment]
Thanks for the response.
Yes of Course!

Traceback (most recent call last):
  File "D:\projects\desk-projects\cisco\venv\lib\site-packages\paramiko\channel.py", line 699,
in recv
    out = self.in_buffer.read(nbytes, self.timeout)
  File "D:\projects\desk-projects\cisco\venv\lib\site-packages\paramiko\buffered_pipe.py", line 164, in read
    raise PipeTimeout()
paramiko.buffered_pipe.PipeTimeout

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\projects\desk-projects\cisco\venv\lib\site-packages\netmiko\base_connection.py", line 573, in _read_channel_expect
    new_data = self.remote_conn.recv(MAX_BUFFER)
  File "D:\projects\desk-projects\cisco\venv\lib\site-packages\paramiko\channel.py", line 701,
in recv
    raise socket.timeout()
socket.timeout

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "d:/projects/desk-projects/cisco/test.py", line 15, in <module>
    connect.send_config_set(config_set)
  File "D:\projects\desk-projects\cisco\venv\lib\site-packages\netmiko\base_connection.py", line 1876, in send_config_set
    output += self.config_mode(*cfg_mode_args)
  File "D:\projects\desk-projects\cisco\venv\lib\site-packages\netmiko\cisco_base_connection.py", line 48, in config_mode
    return super().config_mode(
  File "D:\projects\desk-projects\cisco\venv\lib\site-packages\netmiko\base_connection.py", line 1764, in config_mode
    output += self.read_until_pattern(pattern=pattern, re_flags=re_flags)
  File "D:\projects\desk-projects\cisco\venv\lib\site-packages\netmiko\base_connection.py", line 655, in read_until_pattern
    return self._read_channel_expect(*args, **kwargs)
  File "D:\projects\desk-projects\cisco\venv\lib\site-packages\netmiko\base_connection.py", line 583, in _read_channel_expect
    raise NetmikoTimeoutException(
netmiko.ssh_exception.NetmikoTimeoutException: Timed-out reading channel, data not available.

This is all the error.
actually it happen only with changing hostname multiple times. i tried to change the vlan 5 times in the row and it worked very fast and perfectly. EXP:

for i in range(1, 5):
    config_set = [f'vlan {i}']
    connect.send_config_set(config_set)

but for changing hostname it's work only once or twice and giving me that error.

**Comment by ngergis** [comment]
I have same problem but when I set command `hostname <new-hostname>` with big list of commands after fews commands it fails with same error.

I think it is a bug but I have found out workaround that might help. You need to explicitly call function `set_base_prompt()` after every time you change hostname, so it considers new prompt with the new hostname

I have tried your code, and after adding this line it works:

```
device_info = {my info goes here}
connect = ConnectHandler(**device_info)

for i in range(5):
      config_set = [f'hostname test{i}']
      connect.send_config_set(config_set)
      connect.set_base_prompt()

```

I hope that helps!

**Comment by ktbyers** [comment]
@xSerioUsx78 You failed to go into configuration mode for some reason. Is your user at privilege 15 (i.e. do you need to call `enable()`. You could look at the Netmiko session_log to find out more information why it fails.

**Comment by xSerioUsx78** [comment]
Thank you for your helps @ngergis @ktbyers, today i'll try this.
i figured out a solution before this. actually i just keep it in config mode and when i want to send command in outside of config mode i exit manually and it's work very well and fast.
thank your for your effort and the perfect library! @ktbyers

**Comment by net-nanda** [comment]
I was having the same problem on the Juniper platform. I was sending the whole configuration. In my case, config_set was of type STRING. For example 
config_set = line1
line2
line3
.
.
line N

I converted config_set into type LIST. 
commands = config_set.readlines()
connect.send_config_set(commands)

Hope this helps you as well.

**Comment by ktbyers** [comment]
@net-nanda Can you open a new issue and post your code and exception. this is a very generic exception.
